pageserver/src/branches.rs

```diff
@@ -272,6 +272,16 @@ pub(crate) fn create_branch(
             .get_last_record_lsn();
         info!("branching at end of WAL: {}", end_of_wal);
         startpoint.lsn = end_of_wal;
+    } else if !startpoint.lsn.is_aligned() {
+        info!(
```

Comment:
Why not just align LSN instead and allow to create branch?
And that was point of the test -- LSN's < ~16M would crash pageserver thread on basebackup